RecoTracker/SingleTrackPattern/src/CRackTrajectoryBuilder.cc

```diff
@@ -412,152 +412,89 @@ CRackTrajectoryBuilder::SortHits(const SiStripRecHit2DCollection &collstereo,
   
   yref = (seed_plus) ? yMin : yMax;
   
-  if ((&collpixel)!=0){
-    SiPixelRecHitCollection::DataContainer::const_iterator ipix;
-    for(ipix=collpixel.data().begin();ipix!=collpixel.data().end();ipix++){
-      float ych= RHBuilder->build(&(*ipix))->globalPosition().y();
-      if ((seed_plus && (ych<yref)) || (!(seed_plus) && (ych>yref)))
-	allHits.push_back(&(*ipix));
-    }
-  } 
-  
+  SiPixelRecHitCollection::DataContainer::const_iterator ipix;
+  for(ipix=collpixel.data().begin();ipix!=collpixel.data().end();ipix++){
+    float ych= RHBuilder->build(&(*ipix))->globalPosition().y();
+    if ((seed_plus && (ych<yref)) || (!(seed_plus) && (ych>yref)))
+      allHits.push_back(&(*ipix));
+  }
 
   if (useMatchedHits) // use matched
     {
         //add the matched hits ...
       SiStripMatchedRecHit2DCollection::DataContainer::const_iterator istripm;
 
-      if ((&collmatched)!=0){
-	for(istripm=collmatched.data().begin();istripm!=collmatched.data().end();istripm++){
-	  float ych= RHBuilder->build(&(*istripm))->globalPosition().y();
-
-	  int cDetId=istripm->geographicalId().rawId();
-	  bool noSeedDet = ( detIDSeedMatched.end() == find (detIDSeedMatched.begin(), detIDSeedMatched.end(), cDetId ) ) ;
-
-	  if ( noSeedDet )
+      for(istripm=collmatched.data().begin();istripm!=collmatched.data().end();istripm++){
+	float ych= RHBuilder->build(&(*istripm))->globalPosition().y();
+	
+	int cDetId=istripm->geographicalId().rawId();
+	bool noSeedDet = ( detIDSeedMatched.end() == find (detIDSeedMatched.begin(), detIDSeedMatched.end(), cDetId ) ) ;
+	
+	if ( noSeedDet )
 	  if ((seed_plus && (ych<yref)) || (!(seed_plus) && (ych>yref)))
-	    {
```

Comment:
I'm not a big fan of dropping braces for one-line if/else.
It's easy to get things pretty wrong this way if any edits are to be made around these lines